Update kubernetes.md (#468)
akka/akka-management: docs/src/main/paradox/discovery/kubernetes.md

## Kubernetes API

> Kubernetes API method isn't really a good choice as the default service discovery method. It really only makes sense for cluster bootstrap. Most Kubernetes deployments should use [`akka-dns`](https://doc.akka.io/docs/akka/current/discovery/index.html#discovery-method-dns) as the default `akka.discovery.method`, which will use DNS SRV lookups for service-to-service discovery.

Reviewer: I feel like with this change, the page starts off sounding like it's discouraging people from using it at all. I think the original paragraph still applies. This caveat might be better placed as a [callout](https://developer.lightbend.com/docs/paradox/current/directives/callouts.html#note-callout) in the usage section, maybe by rewording the current bullet list down below on line 19.

Author: But in that case we have to explain how to use it as discovery method, so the following text still applies, isn't it?

```
akka.discovery {
  method = kubernetes-api
}
```

The original text could be updated just adding the "note callout" with the advice to use for cluster bootstrap. Am I misunderstanding something?

Related to the inline files:

* I don't know if the inline yaml examples are outdated. They are similar (but more generic) to those in [cluster bootstrap](https://github.com/akka/akka-management/blob/master/integration-test/kubernetes-api/kubernetes/akka-cluster.yml), maybe the same file could be used with the appropriate annotations above the code.

* The other inline text is just some parts of [reference.conf](https://github.com/akka/akka-management/blob/master/discovery-kubernetes-api/src/main/resources/reference.conf), there is not any example with that level of detail right now, because the examples don't override the default values.

Reviewer: The terminology can be a little confusing. "Discovery method" means an implementation of the Service Discovery API that can be used to lookup addresses given a particular service and port name. There are multiple use cases for using it, which have different needs:

* Cluster Bootstrap is for one node of a service finding its peers, and needs to resolve directly to the pod IPs
* Other cases typically are for one service to find another service, and often the name resolves to the address of a load balancer

So I'm suggesting that it's better to introduce the Kubernetes discovery method by saying what it _is_ good for (cluster bootstrap) rather than what it _isn't_ good for.

Reviewer: #501 rewrote this paragraph entirely, so I think if you take this change out and rebase again, the rest of the changes should be good to merge

Reviewer: Hmm actually some of the other details have also changed in the latest version so it will need a closer look.

Author: I think with #501 this PR is outdated, so I've merely changed the file with minor changes after rebase it with master.

> And configure it to be used as default discovery implementation in your `application.conf`:

Reviewer: This line needs to be updated as well, since it is no longer used as the default, only for cluster bootstrap.

Reviewer: ```suggestion
And configure it to be used for cluster bootstrap in your `application.conf`:
```

* Set `akka.management.cluster.bootstrap.contact-point-discovery.discovery-method` to `kubernetes-api`

> * For bootstrap it is recommended to set the service discovery method via `akka.management.cluster.bootstrap.contact-point-discovery.discovery-method` rather then overriding the global service discovery mechanism with `akka.discovery.method`

Reviewer: Maybe this could be explained a little bit more:

```suggestion
* For bootstrap it is recommended to set the service discovery method via `akka.management.cluster.bootstrap.contact-point-discovery.discovery-method` rather then overriding the global service discovery mechanism with `akka.discovery.method`
* Most Kubernetes deployments should use [`akka-dns`](https://doc.akka.io/docs/akka/current/discovery/index.html#discovery-method-dns) as the default `akka.discovery.method`, which will use DNS SRV lookups for service-to-service discovery
```

```
akka.discovery {
  kubernetes-api {
    # Namespace discovery path
    #
    # If this path doesn't exist, the namespace will default to "default".
    pod-namespace-path = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
  
    # Namespace to query for pods.
    #
    # Set this value to a specific string to override discovering the namespace using pod-namespace-path.

>     pod-namespace = "<pod-namespace>"

Reviewer 2: :+1:
